ls1intum/Artemis
# `Lectures`: Filter lectures on course management page

### Checklist
#### General



#### Client

### Motivation and Context


It allows instructors to filter their lectures based on the start and end dates which can be useful if the course has many lectures and the instructor wants to reduce clutter by filtering out past lectures or quickly find a lecture that is currently in progress to add an attachment. It also allows instructors to find lectures for which the start or end date(s) are undefined.



### Description


Added a Filter button with a dropdown menu and 4 options:
- Past: Filters lectures for which the end date is before the current local time.
- Current: Filters lectures for which the start date is same or before the current local time and the end date is after the current time (i.e. lectures currently in progress)
- Future: Filters lectures for which the start date is after the current local time.
- Unspecified Date(s): Filters Lectures for which the start or end date is undefined.

### Steps for Testing

Prerequisites:
- Instructor

1. Log in to Artemis
2. Navigate to Course Management page
3. Click on Lectures
4. Create lectures with start dates in the past, current, and future, and some without start/end dates specified.
5. Use the filter dropdown menu to ensure that the appropriate lectures are filtered out.

### Review Progress





#### Code Review
#### Manual Tests

### Screenshots



![image](https://user-images.githubusercontent.com/33063132/171255035-24f6facd-51d2-4a7c-8a32-92937ccded3d.png)
![image](https://user-images.githubusercontent.com/33063132/171255132-6fde2b07-22e2-4c54-89b3-e4cf728e8447.png)
